src/main/java/de/tum/in/www1/artemis/service/dataexport/DataExportExerciseCreationService.java

```diff
@@ -228,32 +281,56 @@ private void storeTextSubmissionContent(TextSubmission textSubmission, Path outp
         }
     }
 
-    private void createResultsAndComplaintFiles(Submission submission, Path outputDir) throws IOException {
+    /**
+     * Creates a txt file containing the results with the score, the number of passed test cases if it is a programming exercise
+     * and the feedbacks (both manual and automatic).
+     *
+     * @param submission                                 the submission for which the results should be stored
+     * @param outputDir                                  the directory in which the results should be stored
+     * @param user                                       the user for which the export should be created
+     * @param programmingExerciseBeforeAssessmentDueDate whether the programming exercise is before the assessment due date
+     * @param isInstructor                               whether the user is an instructor in the course the exercise belongs to
+     * @throws IOException if the file cannot be written
+     */
+    private void createResultsAndComplaintFiles(Submission submission, Path outputDir, User user, boolean programmingExerciseBeforeAssessmentDueDate, boolean isInstructor)
+            throws IOException {
         StringBuilder resultScoreAndFeedbacks = new StringBuilder();
         for (var result : submission.getResults()) {
             if (result != null) {
+                // Do not include the results if the assessment due date is in the future and the assessment is not automatic and the user is not an instructor
+                // we only consider progrmming exercises here because for other exercises we this method is not called if the assessment due date is in the future
```

Comment:
```suggestion
                // we only consider programming exercises here because for other exercises this method is not called if the assessment due date is in the future
```
typos